python/tests/expressions/test_expressions.py

```diff
@@ -915,6 +922,96 @@ def test_string_argument_unbound_set():
     assert In("a", {"b", "c"}) == In(Reference("a"), {"b", "c"})
 
 
+@pytest.fixture
+def int_schema() -> Schema:
+    return Schema(NestedField(field_id=1, name="a", field_type=IntegerType(), required=False))
+
+
+@pytest.fixture
+def above_int_max() -> Literal[int]:
+    return IntAboveMax()
+
+
+@pytest.fixture
+def below_int_min() -> Literal[int]:
+    return IntBelowMin()
+
+
+def test_above_int_bounds_equal_to(int_schema: Schema, above_int_max, below_int_min) -> None:
+    assert EqualTo("a", above_int_max).bind(int_schema) is AlwaysFalse()
+    assert EqualTo("a", below_int_min).bind(int_schema) is AlwaysFalse()
```

Comment:
I don't think this tests everything that we need to. This does test that `AboveMax` and `BelowMin` are rewritten, but it doesn't test that they are correctly detected during binding. I think you can fix it by changing the fixtures:

```python
@pytest.fixture
def above_int_max() -> Literal[int]:
    return LongLiteral(IntegerType.max + 1)
```

Reply:
Good one, I've updated the tests to return `Literals` instead.